### PR TITLE
refactor: Use parse-path to detect file paths

### PR DESCRIPTION
**Pull request by ffflorian**

### CODE REVIEW

File: src/Swaxios.ts [hunk 1]
```diff
@@ -150,8 +151,9 @@ async function checkOutputDirectory(outputDirectory: string, forceDeletion?: boo
 
 export async function writeClient(inputFile: string, outputDirectory: string, forceDeletion?: boolean): Promise<void> {
   await checkOutputDirectory(outputDirectory, forceDeletion);
-  const isUrl = /^(https?|ftps?):\/\//.test(inputFile);
-  const swaggerJson = isUrl ? await readInputURL(inputFile) : await readInputFile(inputFile);
+  const parsedPath = parsePath(inputFile);
+  const isFile = parsedPath.protocol === 'file';
```
Comment (bennyn) [inline src/Swaxios.ts]: As far as I can see "parse-path" is used here to check if a path is a file. Can't this be checked with plain [stats.isFile()](https://nodejs.org/docs/latest/api/fs.html#fs_stats_isfile)?

Comment (ffflorian) [reply src/Swaxios.ts]: Unfortunately not, when you run `stats.isFile` on a URL you'll get the same exception as if the file was not found:

```js
fs.statSync('https://example.com').isFile();

[Error: ENOENT: no such file or directory, stat 'https://example.com'] {
  errno: -2,
  code: 'ENOENT',
  syscall: 'stat',
  path: 'https://example.com'
}
```

```js
fs.statSync('/bla').isFile();

[Error: ENOENT: no such file or directory, stat '/bla'] {
  errno: -2,
  code: 'ENOENT',
  syscall: 'stat',
  path: '/bla'
}
```

So we still don't know if it's a file or a URL.

The problem with the currrent regex is that it doesn't accept urls like `localhost/swagger.json` or 'localhost:1234/swagger.json`.

Comment (bennyn) [reply src/Swaxios.ts]: Ok, when this is a requirement for you, then please add tests with these values and then we can merge it.

**Comment by ffflorian** [comment]
It's not that important 😀

**Pull request closed without merging**